packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -742,7 +744,7 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
       const normalizedAcceptedUrls = new Set<string>()
       for (const { url, start, end } of acceptedUrls) {
         const [normalized] = await moduleGraph.resolveUrl(
-          toAbsoluteUrl(url),
+          isVirtualModule(url) ? url : toAbsoluteUrl(url),
```

Comment:
I'm not really sure if this is the right fix. IIUC this captures `virtual:*` only, but not all virtual modules need to start with `virtual:`, they could be e.g. `framework:*`.

Maybe we need to check for relative paths only and run `toAbsoluteUrl` for them?

Reply:
That makes sense. Moreover, I think it's good to remove `isVirtualModule`, this function is only used in https://github.com/vitejs/vite/blob/229c5925c8c84709b8e06e8092a255cb820dafc9/packages/vite/src/node/plugins/importMetaGlob.ts#L357-L358 which I think we can use `!path.isAbsolute(id)` instead (it should work, CI passed locally :P )